Content.Client/Tabletop/TabletopSystem.cs

```diff
@@ -0,0 +1,303 @@
+            SubscribeLocalEvent<TabletopDraggableComponent, ComponentHandleState>(HandleComponentState);
+        }
+
+        public override void Update(float frameTime)
+        {
+            // If there is no player entity, return
+            if (_playerManager.LocalPlayer is not { ControlledEntity: { } playerEntity }) return;
+
+            if (StunnedOrNoHands(playerEntity))
+            {
+                StopDragging();
+            }
+
+            if (!CanSeeTable(playerEntity, _table))
+            {
+                StopDragging();
+                _window?.Close();
+                return;
+            }
+
+            // If no entity is being dragged or no viewport is clicked, return
+            if (_draggedEntity == null || _viewport == null) return;
+
+            // Make sure the dragged entity has a draggable component
+            if (!_draggedEntity.TryGetComponent<TabletopDraggableComponent>(out var draggableComponent)) return;
+
+            // If the dragged entity has another dragging player, drop the item
+            // This should happen if the local player is dragging an item, and another player grabs it out of their hand
+            if (draggableComponent.DraggingPlayer != null &&
+                draggableComponent.DraggingPlayer != _playerManager.LocalPlayer?.Session.UserId)
+            {
+                StopDragging(false);
+                return;
+            }
+
+            // Map mouse position to EntityCoordinates
+            var coords = _viewport.ScreenToMap(_inputManager.MouseScreenPosition.Position);
+
+            // Clamp coordinates to viewport
+            var clampedCoords = ClampPositionToViewport(coords, _viewport);
+            if (clampedCoords.Equals(MapCoordinates.Nullspace)) return;
+
+            // Move the entity locally every update
+            _draggedEntity.Transform.LocalPosition = clampedCoords.Position;
+
+            // Increment total time passed
+            _timePassed += frameTime;
+
+            // Only send new position to server when Delay is reached
+            if (_timePassed >= Delay)
+            {
+                RaiseNetworkEvent(new TabletopMoveEvent(_draggedEntity.Uid, clampedCoords));
+                _timePassed = 0f;
+            }
+        }
+
+        #region Event handlers
+
+        /**
+         * <summary>
+         * Runs when the player presses the "Play Game" verb on a tabletop game.
+         * Opens a viewport where they can then play the game.
+         * </summary>
+         */
+        private void OnTabletopPlay(TabletopPlayEvent msg)
+        {
+            _table = EntityManager.GetEntity(msg.TableUid);
+
+            // Get the camera entity that the server has created for us
+            var camera = EntityManager.GetEntity(msg.CameraUid);
+
+            if (!camera.TryGetComponent<EyeComponent>(out var eyeComponent))
+            {
+                throw new Exception("Camera does not have EyeComponent.");
+            }
```

Comment:
Just do TryGetComponent directly from ComponentManager. Also if you really want it to throw you should just do GetComponent but it should just handle it instead IMO.